src/e2e/liquidity_gap.rs

```diff
@@ -0,0 +1,198 @@
+        invariant
+            .swap(pool_key, true, swap_amount, true, target_sqrt_price)
+            .unwrap();
+
+        let pool = invariant
+            .get_pool(pool_key.token_x, pool_key.token_y, fee_tier)
+            .unwrap();
+
+        let expected_price = calculate_sqrt_price(-10).unwrap();
+        let expected_y_amount_out = U256::from(9999);
+        let liquidity = Liquidity::from_integer(20_006_000);
+        let lower_tick_index = -10;
+
+        assert_eq!(pool.liquidity, liquidity);
+        assert_eq!(pool.current_tick_index, lower_tick_index);
+        assert_eq!(pool.sqrt_price, expected_price);
+
+        let user_x = token_x.balance_of(&caller);
+        let user_y = token_y.balance_of(&caller);
+
+        let dex_x_after = token_x.balance_of(invariant.address());
+        let dex_y_after = token_y.balance_of(invariant.address());
+
+        let delta_dex_x = dex_x_after - dex_x_before;
+        let delta_dex_y = dex_y_before - dex_y_after;
+
+        assert_eq!(user_x, U256::from(0));
+        assert_eq!(user_y, expected_y_amount_out);
+        assert_eq!(delta_dex_x, swap_amount.get());
+        assert_eq!(delta_dex_y, expected_y_amount_out);
+        assert_eq!(
+            pool.fee_growth_global_x,
+            FeeGrowth::new(U128::from(29991002699190242927121u128))
+        );
+        assert_eq!(pool.fee_growth_global_y, FeeGrowth::new(U128::from(0)));
+        assert_eq!(pool.fee_protocol_token_x, TokenAmount::new(U256::from(1)));
+        assert_eq!(pool.fee_protocol_token_y, TokenAmount::new(U256::from(0)));
+    }
+    // No gain swap
+    {
+        let slippage = SqrtPrice::new(U128::from(MIN_SQRT_PRICE));
+        let swap_amount = TokenAmount::new(U256::from(1));
+        let result = invariant.swap(pool_key, true, swap_amount, true, slippage);
+        assert_eq!(result, Err(InvariantError::NoGainSwap));
+    }
+    // Should skip gap and then swap
+    {
+        test_env::set_caller(deployer);
+        let lower_tick_after_swap = -90;
+        let upper_tick_after_swap = -50;
+        let liquidity_delta = Liquidity::from_integer(20008000);
+
+        token_x.mint(&deployer, &liquidity_delta.get());
+        token_y.mint(&deployer, &liquidity_delta.get());
+
+        token_x.approve(invariant.address(), &liquidity_delta.get());
+        token_y.approve(invariant.address(), &liquidity_delta.get());
+
+        let pool_before = invariant
+            .get_pool(pool_key.token_x, pool_key.token_y, fee_tier)
+            .unwrap();
+
+        let slippage_limit_lower = pool_before.sqrt_price;
+        let slippage_limit_upper = pool_before.sqrt_price;
+
+        invariant
+            .create_position(
+                pool_key,
+                lower_tick_after_swap,
+                upper_tick_after_swap,
+                liquidity_delta,
+                slippage_limit_lower,
+                slippage_limit_upper,
+            )
+            .unwrap();
+
+        let caller = test_env::get_account(1);
```

Comment:
"// Should skip gap and then swap" start here. Above is creating seconds position